src/components/kubectl/port-forward.ts

```diff
@@ -0,0 +1,153 @@
+
+        try {
+            const podNames:string[] = pods.map((podObj) => podObj.metadata.name);
+            podSelection = await host.showQuickPick(
+                podNames,
+                { placeHolder: "Select a pod to port-forward to" }
+            );
+        } catch (e) {
+            host.showErrorMessage("Error while selecting pod for port-forward");
+            throw e;
+        }
+
+        if (podSelection === undefined) {
+            host.showErrorMessage("Error while selecting pod for port-forward");
+            return;
+        }
+
+        const targetPort = await promptForPort(podSelection);
+        portForwardToPod(podSelection, Number(targetPort));
+    }
+}
+
+/**
+ * Given a pod name, prompts the user on what port to port-forward to, and validates numeric input.
+ * @param podSelection The pod to port-forward to.
+ */
+async function promptForPort (podSelection: string) {
+    let targetPort: string;
+
+    try {
+        targetPort = await host.showInputBox(<QuickPickOptions>{
+            placeHolder: "8001",
+            prompt: `The numeric port to forward to on pod ${podSelection}`,
+            validateInput: (value) => {
+                if (Number(value) && Number(value) <= MAX_PORT_COUNT) {
+                    return undefined;
+                }
+
+                return `Invalid port. Please enter a valid numerical port:  1 – ${MAX_PORT_COUNT}`;
+            }
+        });
+    } catch (e) {
+        host.showErrorMessage("Could not validate on input port");
+    }
+
+    return targetPort;
+}
+
+/**
+ * Returns one or all available port-forwardable pods.
+ * Checks the open document and returns a pod name, if it can find one.
+ */
+async function findPortForwardablePods () : Promise<PortForwardFindPodsResult> {
+    let kindFromEditor = tryFindKindNameFromEditor();
+    let kind, podName;
+
+    // Find the pod type from the open editor.
+    if (kindFromEditor !== null) {
+        let parts = kindFromEditor.split('/');
+        kind = parts[0];
+        podName = parts[1];
+
+        // Not a pod type, so not port-forwardable, fallback to looking
+        // up all pods.
+        if (kind !== 'pods') {
+            return await findAllPods();
+        }
+
+        return <PortForwardFindPodsResult>{
+            succeeded: true,
+            pods: [podName],
+            fromOpenDocument: true
+        };
+    }
+
+    return await findAllPods() as PortForwardFindPodsResult;
+}
+
+/**
+ * Invokes kubectl port-forward.
+ * @param podName The pod name.
+ * @param targetPort The target port to forward to.
+ * @param localPort (Optional) local port. If not provided, an unbound port is used.
+ */
+export async function portForwardToPod (podName: string, targetPort: number, localPort?: number) {
+    console.log(`port forwarding to pod ${podName} at port ${targetPort}`);
+
+    let usedPort;
+    if (!localPort) {
+        usedPort = await portFinder.getPortPromise({
+            port: 10000
```

Comment:
Why is the local port always 10000?
I would expect:
- either the default `kubectl port-forward` experience, where if you only pass one port this will be the one used both locally and remote
- to have two steps in starting the port forwarding, one asking the local port, the other asking the remote port.

Either way, I would like to see the ability to specify the local port,

What do you think?